pkg/kv/kvserver/store_snapshot.go

```diff
@@ -104,13 +104,11 @@ func (l loggingIncomingSnapshotStream) Send(
 	ctx context.Context, resp *kvserverpb.SnapshotResponse,
 ) error {
 	err := l.stream.Send(resp)
-	if err != nil && ctx.Err() != nil {
+	if err != nil {
 		// Log trace of incoming snapshot on context cancellation (e.g.
 		// times out or caller goes away).
-		if sp := tracing.SpanFromContext(ctx); sp != nil && !sp.IsNoop() {
-			log.Infof(ctx, "incoming snapshot stream response send failed with error: %s\ntrace:\n%s",
-				err, sp.GetConfiguredRecording())
-		}
+		log.Infof(ctx, "incoming snapshot stream response send failed with error: %s\ntrace:\n%s",
```

Comment:
Q for here and below: in @adityamaru's recent PR about tracing there was extra work done to ensure the trace is not printed if it is empty. Do you care about that here too?